neutron_lbaas_dashboard/static/app/core/openstack-service-api/barbican.service.js

```diff
@@ -34,10 +35,12 @@
    * @returns The barbican service API.
    */
 
-  function barbicanAPI(apiService, toastService) {
+  function barbicanAPI($q, apiService, toastService) {
```

Comment:
Is it possible not to redefine this function declaration?

Reply:
Tested, however, we have to depend on $q to pass promise.

Reply:
$q is a global variable?

Reply:
yes, is a service, similar to C/C++ <stdio.h>, or python's `import json`